design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in default and kube-system namespaces, and all cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-resources=true
+```
+
+#### all namespaced resources + no cluster resources
+The following commands all mean backup all namespace scope resources and no cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-cluster-resources=false
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=*
+	--include-cluster-resources=false
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
+	--include-cluster-resources=false
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-cluster-resources=false
+```
+
+#### all namespaced resources + only related cluster resources
+The following commands all mean backup all resources in all namespaces, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
+```
+
+``` bash
+velero backup create <backup-name>
+```
+
+#### all namespaced resources + some additional cluster resources
+The following commands all mean backup all namespace scope resources, and related cluster resources, plus all PersistentVolume resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-cluster-scope-resources=PersistentVolume
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=*
+	--include-cluster-scope-resources=PersistentVolume
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
+	--include-cluster-scope-resources=PersistentVolume
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-cluster-scope-resources=PersistentVolume
+```
+
+#### all namespaced resources + all cluster resources
+The following commands have the same meaning: backup all namespace scope resources, and all cluster scope resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
+	--include-cluster-scope-resources=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-namespaced-resources=*
+	--include-cluster-resources=true
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=*
+	--include-cluster-scope-resources=*
```

Comment:
--include-namespaced-resources=*

Reply:
I'm not sure if this is needed anymore, b/c  when --include-namespaced-resources is not set it means include everything.

Reply:
@reasonerjt That's the old behavior. We agreed earlier to require this for a couple reasons:
1) consistency with include-clsuter-scope-resources
2) Allow not including any namespaced resources.

Reply:
@sseago 
After a discussion with Daniel, we think when `--include-namespaced-resources` is not set, it means including every namespace resource; when `--include-cluster-scope-resources` is not set, it means including related cluster scope resources, which is the same with existing `--include-cluster-resources`.

To not include any namespace resources, we set `--exclude-namespaced-resources=*`.